Shear/HistoryStorageType.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  HistoryStorageType.swift
+//  Shear
+//
+//  Created by Andreas Kompanez on 23.02.20.
+//  Copyright © 2020 Andreas Kompanez. All rights reserved.
+//
+
+import Foundation
+
+enum HistoryStorageStoreError: Error {
+    case undefined
+    case duplicate
+}
+
+typealias HistoryStorageStoreHandler = (_ error: HistoryStorageStoreError?, _ item: HistoryItem) -> Void
+
+protocol HistoryStorageType {
+    var count: Int { get }
+    
+    func store(item: HistoryItem, handler: @escaping HistoryStorageStoreHandler)
+    
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/kmpnz/Shear/blob/895e50bbb85826ea77037e4a1c1e5468d960c4fc/Shear/HistoryStorageType.swift#L22"></span>Lines should not have trailing whitespace.<br />
<code>trailing_whitespace</code> <code>HistoryStorageType.swift:22</code></del></td>
    </tr>
  </tbody>
</table>